arabic/unpacked/arabic.js

```diff
@@ -0,0 +1,448 @@
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+MathJax.Hub.Register.StartupHook('Arabic TeX Startup', function () {
+  MathJax.Extension.Arabic.config = MathJax.Hub.CombineConfig('Arabic', {
+    identifiersMap: {
+      // Variable names
+      'a': 'أ',
+      'b': 'ب',  // TODO: Consider using Arabic letter dotless beh 0x66e instead
+      'c': 'جـ',  // Suffixed with Unicode Arabic Tatweel 0x0640
+      'x': 'س',
+      'y': 'ص',
+      'z': 'ع',
+      'n': 'ن',
+
+      // Function names
+      'f': 'ق',  // TODO: Consider using dotless qaf (ٯ) instead
+      'g': 'جـ',  // With Unicode Arabic Tatweel 0x0640
+      'h': 'هـ',  // With Unicode Arabic Tatweel 0x0640
+
+      // Mixed use
+      'k': 'ك',
+      'r': 'ر',
+      't': 'ت',
+      'd': 'د',  // Function, variable and (dx)
+      'e': 'هـ',  // With Unicode Arabic Tatweel 0x0640
+      'm': 'م',
+      'l': 'ل'
+    }
+  });
+});
+
+MathJax.Hub.Register.StartupHook('HTML-CSS Jax Require', function () {
+  MathJax.Hub.Config({
+    'HTML-CSS': {
+      styles: {
+        '.MathJax .mfliph': {
+          'display': 'inline-block !important',
+          '-moz-transform': 'scaleX(-1)',
+          '-webkit-transform': 'scaleX(-1)',
+          '-o-transform': 'scaleX(-1)',
+          'transform': 'scaleX(-1)',
+          '-ms-filter': 'fliph',
+          'filter': 'fliph'
+        },
+        '.MathJax .mar': {
+          'font-style': 'normal !important'
+        },
+        '.MathJax .mar > span': {
+          'font-style': 'normal !important'
+        }
+      }
+    }
+  });
+});
+
+
+MathJax.Hub.Register.StartupHook('HTML-CSS Jax Ready', function () {
+  MathJax.Hub.Register.StartupHook('Arabic TeX Ready', function () {
+    var MML = MathJax.ElementJax.mml;
+
+    var makeElementFlippable = function (name) {
+      var originalToHTML = MML[name].prototype.toHTML;
+
+      MML[name].Augment({
+        toHTML: function () {
+          var element = originalToHTML.apply(this, arguments);
+
+          if (this.arabicFlipH) {
+            var flipElement = document.createElement('span');
+
+            var className = ' mfliph';  // Keep the leading space
+
+            if ('ar' === this.arabicFontLang) {
+              className += ' mar';  // Keep the leading space
+            }
+
+            flipElement.className = className;
```

Comment:
The code from lines 96 to 102 could be simplified.  There is really no need for a separate variable for this, and the initial space is not needed in line 96 (despite the comment) since there is no previous `className` value to worry about.  You could just do

    flipElement.className = 'mfliph';
    if ('ar' === this.arabicFontLang) flipElement.className += ' mar';